token-metadata/program/src/state/metadata.rs

```diff
@@ -98,104 +102,114 @@ impl Metadata {
             ..
         } = args;
 
-        if let Some(data) = data {
-            if !self.is_mutable {
-                return Err(MetadataError::DataIsImmutable.into());
-            }
+        // only allow update if the token standard is known
+        let token_standard = self
```

Comment:
Since this is an update and the update authority or delegate is signing I think we could just automatically update their token standard for them if it's not set, but that can be a separate PR.

Reply:
Done.